lib/ddtrace/configuration/agent_settings_resolver.rb

```diff
@@ -20,25 +20,47 @@ module Configuration
     class AgentSettingsResolver
       AgentSettings = \
         Struct.new(
+          :adapter,
           :ssl,
           :hostname,
           :port,
+          :uds_path,
           :timeout_seconds,
           :deprecated_for_removal_transport_configuration_proc,
           :deprecated_for_removal_transport_configuration_options
         ) do
           def initialize(
+            adapter:,
             ssl:,
             hostname:,
             port:,
+            uds_path:,
             timeout_seconds:,
             deprecated_for_removal_transport_configuration_proc:,
             deprecated_for_removal_transport_configuration_options:
           )
-            super(ssl, hostname, port, timeout_seconds, deprecated_for_removal_transport_configuration_proc, \
-              deprecated_for_removal_transport_configuration_options)
+            super(
+              adapter,
+              ssl,
+              hostname,
+              port,
+              uds_path,
+              timeout_seconds,
+              deprecated_for_removal_transport_configuration_proc,
+              deprecated_for_removal_transport_configuration_options,
+            )
             freeze
           end
+
+          def merge(**member_values)
+            new_struct = dup
+
+            member_values.each do |member, value|
+              new_struct[member] = value
+            end
+
+            new_struct
+          end
```

Comment:
Interesting! This approach of being able to use this to do "local overrides" of the agent settings before they get passed to the transport is interesting.

May I suggest also calling `freeze` on the `new_struct` that gets returned, to keep the current approach where instances are always immutable?

Reply:
Freeze is a great idea, done!